app/controllers/green_lanes/results_controller.rb

```diff
@@ -2,66 +2,67 @@ module GreenLanes
   class ResultsController < ApplicationController
     include GreenLanesHelper
 
-    before_action :check_green_lanes_enabled,
-                  :disable_switch_service_banner,
-                  :disable_search_form
+    before_action :set_up, only: :show
 
     def show
-      goods_nomenclature = GreenLanes::GoodsNomenclature.find(
-        results_params[:commodity_code],
-        { filter: {
-          geographical_area_id: results_params[:country_of_origin],
-          moving_date: results_params[:moving_date],
-        } },
-        { authorization: TradeTariffFrontend.green_lanes_api_token },
-      )
+      @next_page = determine_next_page
+    end
+
+    private
+
+    def set_up
+      check_green_lanes_enabled
+      disable_switch_service_banner
+      disable_search_form
+      goods_nomenclature
+      set_instance_variables
+      determine_assessments
+    end
```

Comment:
we are mixing many concern here, it may be difficult to maintain.

- We could keep:
      check_green_lanes_enabled
      disable_switch_service_banner
      disable_search_form
      
 in the before filter.
 
- we probably don't need to force the memorization on `goods_nomenclature`

- `determine_assessments` is only called in the initialize, we can just call the assignments in it

Reply:
So this again was part of a slight refactor happy to discuss. actually it would be better to move some of these into the green lane helper?